pysynphot/locations.py

```diff
@@ -168,65 +135,91 @@ def _get_RedLaws():
+                  'future versions of pysynphot.')
+
 
+#Define wavecat file explicitly
+wavecat = get_data_filename('wavecat.dat')
 
+# load the extintion law file names
+_get_RedLaws()
 
 ## This dictionary maps IRAF-specific directory names for synphot
 ## directories into their Unix equivalents.
 #BUG: supports only a single variable in a string
 #............basically this is a weak routine that should be made
 #............more robust
 #BUG: this dictionary should be in a data file
-CONVERTDICT = {'crrefer':rootdir,
-              'crotacomp':os.path.join(rootdir,'comp','ota'),
-              'cracscomp':os.path.join(rootdir,'comp','acs'),
-              'crcalobs':os.path.join(rootdir,'calobs'),
-              'crcalspec':os.path.join(rootdir,'calspec'),
-              'croldcalspec':os.path.join(rootdir,'oldcalspec'),
-              'crcomp':os.path.join(rootdir,'comp'),
-              'crfgs':os.path.join(rootdir,'fgs'),
-              'crfields':os.path.join(rootdir,'fields'),
-              'crmodewave':os.path.join(rootdir,'modewave'),
-              'crcostarcomp':os.path.join(rootdir,'comp','costar'),
-              'cracscomp':os.path.join(rootdir,'comp','acs'),
-              'crfoccomp':os.path.join(rootdir,'comp','foc'),
-              'crfoscomp':os.path.join(rootdir,'comp','fos'),
-              'crfgscomp':os.path.join(rootdir,'comp','fgs'),
-              'crhrscomp':os.path.join(rootdir,'comp','hrs'),
-              'crhspcomp':os.path.join(rootdir,'comp','hsp'),
-              'crotacomp':os.path.join(rootdir,'comp','ota'),
-              'crnicmoscomp':os.path.join(rootdir,'comp','nicmos'),
-              'crnonhstcomp':os.path.join(rootdir,'comp','nonhst'),
-              'crstiscomp':os.path.join(rootdir,'comp','stis'),
-              'crstiscomp':os.path.join(rootdir,'comp','stis'),
-              'crwfc3comp':os.path.join(rootdir,'comp','wfc3'),
-              'crcoscomp':os.path.join(rootdir,'comp','cos'),
-              'coscomp':os.path.join(rootdir,'comp','cos'),
-              'crwave':os.path.join(rootdir,'crwave'),
-              'crwfpccomp':os.path.join(rootdir,'comp','wfpc'),
-              'crwfpc2comp':os.path.join(rootdir,'comp','wfpc2'),
-              'crgrid':os.path.join(rootdir,'grid'),
-              'crgridbz77':os.path.join(rootdir,'grid','bz77'),
-              'crgridgs':os.path.join(rootdir,'grid','gunnstryker'),
-              'crgridjac':os.path.join(rootdir,'grid','jacobi'),
-              'crgridbpgs':os.path.join(rootdir,'grid','bpgs'),
-              'crgridbk':os.path.join(rootdir,'grid','bkmodels'),
-              'crgridk93':os.path.join(rootdir,'grid','k93models'),
-              'crgridagn':os.path.join(rootdir,'grid','agn'),
-              'crgridgalactic':os.path.join(rootdir,'grid','galactic'),
-              'crgridkc96':os.path.join(rootdir,'grid','kc96'),
-              'mtab':os.path.join(rootdir,'mtab'),
+CONVERTDICT = {'crrefer':pysynphot._root(),
```

Comment:
It does not make sense to actually call the `_root()` functions so many times. Better to call it once to set `rootdir` and leave the original codes as-is.